scripts/fetch_olean_cache.sh

```diff
@@ -0,0 +1,8 @@
+set -e
+set -x
+
+archive_url="https://oleanstorage.azureedge.net/mathlib/"
+
+new_git_sha=$(python scripts/look_up_olean_hash.py $lean_file_hash) || exit 0
+echo "equivalent Git sha: $new_git_sha"
+curl "$archive_url$new_git_sha.tar.gz" | tar xz --wildcards *.olean
```

Comment:
Are you sure this will work?  I would expect the olean files to be older than the lean files after this line.
```suggestion
curl "$archive_url$new_git_sha.tar.gz" | tar xz
git reset --hard
```
This should keep the timestamps consistent. ^^

Reply:
The `git reset --hard` is just a defensive precaution, in case the lean files in the tarball are different from the current ones.

Reply:
I'm not sure. It seemed to work in testing but maybe that was a coincidence. Or maybe it actually rebuilt the oleans faster than I expected. Let's see what happens with the recent commit.

I was hesitant to unpack the whole archive. If we get an old version that has extra files in it, those files will appear in the archive that gets uploaded after CI runs. Right now we're only using the archives for oleans, but maybe in the future we'll use them for something else, and maybe they'll have to be accurate.

Reply:
Well, that's a bad sign, it seems to have computed a different file hash? Despite no relevant changes.

Reply:
> If we get an old version

This should never happen, right?  (Unless there is a hash collision, of course.)

> that's a bad sign, it seems to have computed a different file hash?

This is disturbing, I get a different hash locally (but it's the same for both of the revision where github actions has run):
```
$ echo `find src | grep '[.]lean$' | sort | xargs shasum` `shasum leanpkg.toml` "lean --version:" `lean --version` | shasum | head -c 40; echo
74b931efdf2366028ade4066d225496a7829c409
```

Reply:
> This should never happen, right? (Unless there is a hash collision, of course.)

The situation I'm worried about is: we have a commit that doesn't touch any `.lean` files in `src`, but deletes a file somewhere else. This is functionally equivalent to the previous commit, so we get that archive, which means the deleted file gets included in the archive for the new commit. Granted, this is a pretty minor concern.

We could avoid it by only unpacking the `src` directory. (Assuming we'll only ever have `.lean` files in there.)

Reply:
> This is disturbing, I get a different hash locally

And the hash I get is different from both yours and Actions'.

Reply:
This time Actions computed the same hash: look at that speedy build! https://github.com/leanprover-community/mathlib/pull/2094/checks?check_run_id=493673348

Reply:
> We could avoid it by only unpacking the `src` directory. (Assuming we'll only ever have `.lean` files in there.)

The tarballs only include the `src` directory at the moment.

Reply:
...okay then. Not sure why I thought we were including everything!

Reply:
The different output on my machine (Windows, Git Bash) is because it's sorting differently.
```
  0e8cbbd2c21c02531afd72a235438455d3798005 src/category_theory/category/Cat.lean
+ 141e52d0ab9967ae5b3ae08c8b83acf29de772e0 src/category_theory/category/default.lean
  497f81c4f944f253306e2b225df1da2b8276e499 src/category_theory/category/Groupoid.lean
  37d6b51a57fbfc1ad703864c752c6478b6fe85ef src/category_theory/category/Kleisli.lean
  a7e354ecfac6d4d39fb3ad5f28db5dd77b7aa461 src/category_theory/category/Rel.lean
- 141e52d0ab9967ae5b3ae08c8b83acf29de772e0 src/category_theory/category/default.lean
```